modules/openapi-generator/src/main/java/org/openapitools/codegen/TemplateManager.java

```diff
@@ -145,18 +158,32 @@ public Reader getTemplateReader(String name) {
      */
     @Override
     public File write(Map<String, Object> data, String template, File target) throws IOException {
-        String templateContent = this.engineAdapter.compileTemplate(this, data, template);
-        return writeToFile(target.getPath(), templateContent);
+        if (this.engineAdapter.handlesFile(template)) {
+            // Only pass files with valid endings through template engine
+            String templateContent = this.engineAdapter.compileTemplate(this, data, template);
+            return writeToFile(target.getPath(), templateContent);
+        } else {
+            // Do a straight copy of the file if not listed as supported by the template engine.
+            InputStream is;
+            try {
+                // look up the file using the same template resolution logic the adapters would use.
+                String fullTemplatePath = getFullTemplateFile(template);
```

Comment:
This fixes what I believe may have been a bug prior to this change… supporting files may not have resolved using the libraries path rules appropriately.